cmd/rpc.go

```diff
@@ -26,8 +26,15 @@ import (
 
 	xrpc "github.com/minio/minio/cmd/rpc"
 	xnet "github.com/minio/minio/pkg/net"
+	xtime "github.com/minio/minio/pkg/time"
 )
 
+// minRPCRetryTimeout - minimum RPC retry timeout on network failure is one minute.
+const minRPCRetryTimeout = 1 * time.Minute
```

Comment:
I figured we can move this value to 5 seconds and eventually go to 30minutes? - the reason is today even with intermittent network disconnections we lose the ability to bring the disk back into the cluster, culminating in quorum errors even when the disk is back online.

Atleast we need the ability to differentiate between various network errors differently

- i/o timeout should be treated with fewer retry counts i.e server is up but busy we can retry next call quickly.
- no route to host should be treated with higher retry counts i.e server is not reachable
- dial disconnected should be treated with medium level retry counts

What do you think? @balamurugana

Reply:
I am not aware that we have moved the values from 5 seconds to 30 minutes.  Are those values for network failure retry?

Rest of timeouts are different issues which is not related to this PR because it fixes `rpc: use exponential retry timeout between network failure.`

Reply:
Not particular to this PR alone, but I am thinking that 1 minute for all network errors is actually not correct and can lead prematurely thinking that disks offline for a longer period of time.

Reply:
What are values decided for network failures?

Reply:
Not yet decided need to check with @abperiasamy on what could be a better value on different types of values.

Reply:
Mentioned values in the PR were decided with previous discussion.  Once I get new values, i will update accordingly.